### ISSUE TITLE
Make UX & Design for the Quarkus Getting Started V1

### ISSUE THREAD

**Issue by ia3andy** [fabric8-launcher/launcher-quarkus#2]
@jkbrock this issue is to discuss, track and achieve UX for the Quarkus Getting Started V1

- [ ] Design theme
- [ ] Implement theme
- [ ] Validate labels
- [ ] Write content of the "What's next popup"

**Comment by emmanuelbernard** [comment]
What is the "What's next pop up"?

**Comment by emmanuelbernard** [comment]
- [ ] move the list of extensions from model window to putting it below in the main page to improve discovery

Maybe

- [ ] use shortname as the search that should be move atop in the proposed results in autocompletion

**Comment by ia3andy** [comment]
@emmanuelbernard 
> What is the "What's next pop up"?
![image](https://user-images.githubusercontent.com/2223984/59745483-2fa9d900-9275-11e9-8c9f-a3f10d9f674f.png)

**Comment by emmanuelbernard** [comment]
I think we need a small intro similar to this

https://quarkus.io/extensions/

```
Think of Quarkus extensions as your project dependencies. Extensions configure, boot and integrate a framework or technology into your Quarkus application. They also do all of the heavy lifting of providing the right information to GraalVM for your application to compile natively.

Adding an extension is as easy as adding a dependency or use Quarkus tooling.

$ mvn quarkus:list-extensions
$ mvn quarkus:add-extension -Dextensions="groupId:artifactId"

# Or

$ gradle list-extensions
# and add the dependency in your dependencies section


Explore the wide breath of technologies Quarkus applications can be made with.
```

At the top of the page. Maybe it cloud be foldable, once someone has seen it once. that would be nice.

**Comment by ia3andy** [comment]
This is the most recent mock design made by @jkbrock:
https://xd.adobe.com/view/982afb48-3e4b-4724-7901-7ba296b8c404-ee1b/?fullscreen

**Comment by ia3andy** [comment]
Thank you Jason!!

Making the "options" to the side is really nice!

I have 2 feedbacks:
1. I would like the download button to be sticky because it shows the keyboard shortcut, and some could want to download directly
2. I find that having all those contrasted lines (inputs + boxes + separators) makes things overwhelming (though maybe that's just me 😅)

**Comment by ia3andy** [comment]
@emmanuelbernard  said:
> Assume that the list of extensions will be 200 strong and the categories of extensions be up to 20 strong.
Right know we are at 50 ish extensions from memory.

> I still don't understand that the search box is a search box.
With the list becoming 20 strong, we might need:
> - a reminder of what's selected at the top or a way to filter by selected
> - a generate button above the list (grayed as you said)

> The header looks like a Bootstrap project. So not a big fan but indeed let's separate that conversation.

> Above Project Metadata there will be a blurb about extensions.Something similar to

---
Think of Quarkus extensions as your project dependencies. Extensions configure, boot and integrate a framework or technology into your Quarkus application. They also do all of the heavy lifting of providing the right information to GraalVM for your application to compile natively.

Adding an extension is as easy as adding a dependency or use Quarkus tooling.
```
$ mvn quarkus:list-extensions
$ mvn quarkus:add-extension -Dextensions="groupId:artifactId"

# Or

$ gradle list-extensions
# and add the dependency in your dependencies section
```      

Explore the wide breath of technologies Quarkus applications can be made with.

---

**Comment by ia3andy** [comment]
**I don't want to slow this down or start a big discussion on this (if you don't like just say I don't like and we continue):**

I was thinking a bit out of the box on how we could provide a great xp that would be different from what the competition does and be fun to use by devs.

Maybe we could provide a fully automated terminal like experience (http://terminal-in-react.surge.sh/):
```
#  Welcome on the Quarkus project generator v1.0.0
#  We are going to get you through customizing and generating your Quarkus project
#  At anytime, hit alt+enter to generate your project, escape to restart from scratch 
#  Fill those to describe your project, (hit enter to use default value)

    group (com.testproject):
    artifact (test-project):
    version: (1.0.0-SNAPSHOT):
    packageName (org.example):

# Think of Quarkus extensions as your project dependencies. 
# Extensions configure, boot and integrate a framework or technology into your Quarkus application.
# They also do all of the heavy lifting of providing the right information to GraalVM for your application to compile natively.
# Once the project is generated, Adding an extension is as easy as adding a dependency or use Quarkus tooling (guide for maven, gradle).

# a, add --> add an extension with auto-completion
# a, rem --> remove an extension with auto-completion
# l, list --> see all extensions
# s, selected --> see your selection
# At anytime, hit alt+enter to generate your project, escape to restart from scratch 
> ...
```

I know it would be a complete change from what we have, but at least, no one's done it, it's familiar for devs..

**Comment by emmanuelbernard** [comment]
Not a big fan @ia3andy 
The problem with terminal tools is that while they are great for experts, the discovery aspect often suffers. This is no different where the list of extensions would be vastly harder to see, likewise the search engine would not have the autocompletion we are experiencing from the web version.

**Comment by ia3andy** [comment]
@emmanuelbernard yeah, you are right, I was just dreaming a bit too much I guess..

**Comment by ia3andy** [comment]
@emmanuelbernard  though, I've been playing a bit around the idea, and I guess that theme wise, it looks really nice:
![image](https://user-images.githubusercontent.com/2223984/60522134-b7b5c700-9ce8-11e9-81ea-eb04c5296027.png)

We'll discuss that further at the meeting..

**Comment by kenfinnigan** [comment]
I'm liking the latest styling.

Though when the list of extensions grows that might become overwhelming with many boxes?

**Comment by jkbrock** [comment]
@kenfinnigan , not might. It will *definitely* become overwhelming. In most situations, the ideal number of choices is "fewer." 

I think for right now we might just categorize them in the interest of getting this thing launched, and then iterate on better filters as more extensions are added.

**Comment by kenfinnigan** [comment]
sounds reasonable

**Comment by jkbrock** [comment]
@ia3andy and @emmanuelbernard here is a link to a new version of the workflow. I've changed a few things up to try to balance out the idea of Terminal interface with the existing brand.

https://xd.adobe.com/view/873a5241-8d88-4c7c-7eb7-5d05360e89f9-f10b/?fullscreen

**Comment by kenfinnigan** [comment]
I like this!

**Comment by ia3andy** [comment]
@jkbrock indeed, love it!!

**Comment by ia3andy** [comment]
@jkbrock do you have a mobile version too?

**Comment by jkbrock** [comment]
@ia3andy , mobile version forthcoming.

**Comment by jkbrock** [comment]
@ia3andy Here's a mobile version. Let me know if you want to meet tomorrow to discuss

https://xd.adobe.com/view/17231a1b-ef7a-45c3-7a25-c42a1e1bd8a2-6112/

**Comment by ia3andy** [comment]
@jkbrock you did an amazing job :) I am really happy to implement it!

I don't know if I'll be able to do everything for the first version, but I'll try to cover as much as possible.

Thanks !!

**Comment by ia3andy** [comment]
This is for me so I don't loose it: All the Quarkus logo https://design.jboss.org/quarkus/
